examples/fluids/navierstokes.c

```diff
@@ -1013,33 +1029,68 @@ int main(int argc, char **argv) {
   ierr = VecGetSize(Qloc, &lnodes); CHKERRQ(ierr);
   lnodes /= ncompq;
 
-  {
-    // Print grid information
+  // Initialize CEED
+  CeedInit(ceedresource, &ceed);
+  // Set memtype
+  CeedMemType memtypebackend;
+  CeedGetPreferredMemType(ceed, &memtypebackend);
+  // Check memtype compatibility
+  if (!setmemtyperequest)
+    memtyperequested = memtypebackend;
+  else if (!petschavecuda && memtyperequested == CEED_MEM_DEVICE)
+    SETERRQ1(PETSC_COMM_WORLD, PETSC_ERR_SUP_SYS,
+             "PETSc was not built with CUDA. "
+             "Requested MemType CEED_MEM_DEVICE is not supported.", NULL);
+
+  // Set number of 1D nodes and quadrature points
+  numP = degree + 1;
+  numQ = numP + qextra;
+
+  // Print summary
+  if (testChoice == TEST_NONE) {
     CeedInt gdofs, odofs;
     int comm_size;
     char box_faces_str[PETSC_MAX_PATH_LEN] = "NONE";
     ierr = VecGetSize(Q, &gdofs); CHKERRQ(ierr);
     ierr = VecGetLocalSize(Q, &odofs); CHKERRQ(ierr);
+    gnodes = gdofs/ncompq;
     ierr = MPI_Comm_size(comm, &comm_size); CHKERRQ(ierr);
     ierr = PetscOptionsGetString(NULL, NULL, "-dm_plex_box_faces", box_faces_str,
                                  sizeof(box_faces_str), NULL); CHKERRQ(ierr);
-    if (testChoice == TEST_NONE) {
-      ierr = PetscPrintf(comm, "Global FEM dofs: %D (%D owned) on %d rank(s)\n",
-                         gdofs, odofs, comm_size); CHKERRQ(ierr);
-      ierr = PetscPrintf(comm, "Local FEM nodes: %D\n", lnodes); CHKERRQ(ierr);
-      ierr = PetscPrintf(comm, "dm_plex_box_faces: %s\n", box_faces_str);
-      CHKERRQ(ierr);
-    }
+    const char *usedresource;
+    CeedGetResource(ceed, &usedresource);
+
+    ierr = PetscPrintf(comm,
+                       "\n-- libCEED Navier-Stokes solver + PETSc --\n"
+                       "  rank(s):                             : %d\n"
+                       "  PETSc:\n"
+                       "    dm_plex_box_faces:                 : %s\n"
```

Comment:
A few small thoughts,
- The title is a bit different in style from the solilds example, though that's no big deal to me
"\n-- Elasticity Example - libCEED + PETSc --\n"

- Is there a better name for dm_plex_box_faces? Box Faces?

- Do we want to include the problem type and stabilization type?

Overall I like this a lot better.

Reply:
Done in [6f55dfd](https://github.com/CEED/libCEED/pull/516/commits/6f55dfd5a730b78d8f668c0a3f4f8ca678245335) which unfortunately also includes some changes from master after I tried to reduce the number of commits (my bad). 

Please let me know if the summary print statement looks good now. Thanks